### PR TITLE
feat: Add exponential backoff and 3 retries on 429 error

### LINKED ISSUE #713: Add retries to the client for 429 (Too Many Requests) errors
### Issue Summary
[Twilio's recommendation](https://support.twilio.com/hc/en-us/articles/360044308153-Twilio-API-response-Error-429-Too-Many-Requests-
) is that clients implement retries with exponential backoff to the API client to deal with 429 (Too Many Requests errors and link documentation about this from Google, Amazon and Microsoft about implementing this.

However vendor provided API clients (such as this project) should be helping uses by implementing best practice recommendations 'out of the box', so I would expect this project to be providing the recommended backoff functionality.

All of the companies whose documentation you link provide a pre-built mechanism for retrying failed requests in their API clients and SDKs:

- [Google Cloud](https://googleapis.dev/python/google-api-core/latest/retry.html)
- [AWS](https://docs.aws.amazon.com/general/latest/gr/api-retries.html)
- [Microsoft](https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific)

It would be nice if Twilio could do the same. This would save a significant amount of hassle for users, as it would save every user from having to implement the recommendations themselves, again and again. It would reduce the support burden on Twilio, as all users would be getting a correctly implemented exponential backoff out of the box and so wouldn't be asking support about how to go about implementing this.

### Technical details:
* twilio-node version: v3.71.3
* node version: v12.21.0

### PR DESCRIPTION
**Pull request by bobtfish**
# Fixes #713

Add exponential backoff and 3 retries on 429 error per [the recommendations](https://support.twilio.com/hc/en-us/articles/360044308153-Twilio-API-response-Error-429-Too-Many-Requests-)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-node/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: lib/base/RequestClient.js [hunk 1]
```diff
@@ -11,6 +11,26 @@ var Request = require('../http/request');
...
+axios.interceptors.response.use(res => {
+  const { config } = res
+  if (res.status !== 429) {
+    return res
+  }
+
+  const retryCount = (config.retryCount || 0) + 1
+  config.retryCount = retryCount
+  if (retryCount <= 3) {
+    const baseDelay = Math.pow(2, retryNumber) * 100
```
Comment (shwetha-manvinkurke) [inline lib/base/RequestClient.js:23]: Do you mean `retryCount` here?

Comment (thomaschaaf) [reply lib/base/RequestClient.js:23]: @bobtfish did you see this comment?

**Comment by JenniferMah** [comment]
Closing until PR feedback is addressed.

**Comment by bobtfish** [comment]
I was kinda hoping that Twilio would run with this and take it over the line @JenniferMah - every other API client that we use a lot (NSOne, AWS, Google, etc etc) encode their recommended best practices in their API clients rather than leaving every client to implement it for themselves.

**Comment by bobtfish** [comment]
I've added the trivial fixes that were needed in https://github.com/twilio/twilio-node/pull/741

**Pull request closed without merging**